scipy/special/specfun_wrappers.cpp

```diff
@@ -0,0 +1,226 @@
+  return npy_cpack(res.real(), res.imag());
+}
+
+double hypU_wrap(double a, double b, double x) {
+  return special::hypU_wrap(a, b, x);
+}
+
+double hyp1f1_wrap(double a, double b, double x) {
+  return special::hyp1f1_wrap(a, b, x);
+}
+
+int itairy_wrap(double x, double *apt, double *bpt, double *ant, double *bnt) {
+  return special::itairy_wrap(x, apt, bpt, ant, bnt);
+}
+
+double exp1_wrap(double x) { return special::exp1_wrap(x); }
+
+npy_cdouble cexp1_wrap(npy_cdouble z) {
+  std::complex<double> res = special::cexp1_wrap({npy_creal(z), npy_cimag(z)});
+  return npy_cpack(res.real(), res.imag());
+}
+
+double expi_wrap(double x) { return special::expi_wrap(x); }
+
+npy_cdouble cexpi_wrap(npy_cdouble z) {
+  std::complex<double> res = special::cexpi_wrap({npy_creal(z), npy_cimag(z)});
+  return npy_cpack(res.real(), res.imag());
+}
+
+npy_cdouble cerf_wrap(npy_cdouble z) {
+  std::complex<double> res = special::cerf_wrap({npy_creal(z), npy_cimag(z)});
+  return npy_cpack(res.real(), res.imag());
+}
+
+double itstruve0_wrap(double x) { return special::itstruve0_wrap(x); }
+
+double it2struve0_wrap(double x) { return special::it2struve0_wrap(x); }
+
+double itmodstruve0_wrap(double x) { return special::itmodstruve0_wrap(x); }
+
+double ber_wrap(double x) { return special::ber_wrap(x); }
+
+double bei_wrap(double x) { return special::bei_wrap(x); }
+
+double ker_wrap(double x) { return special::ker_wrap(x); }
+
+double kei_wrap(double x) { return special::kei_wrap(x); }
+
+double berp_wrap(double x) { return special::berp_wrap(x); }
+
+double beip_wrap(double x) { return special::beip_wrap(x); }
+
+double kerp_wrap(double x) { return special::kerp_wrap(x); }
+
+double keip_wrap(double x) { return special::keip_wrap(x); }
+
+int kelvin_wrap(double x, npy_cdouble *Be, npy_cdouble *Ke, npy_cdouble *Bep,
+                npy_cdouble *Kep) {
+  return special::kelvin_wrap(x, reinterpret_cast<std::complex<double> *>(Be),
+                              reinterpret_cast<std::complex<double> *>(Ke),
+                              reinterpret_cast<std::complex<double> *>(Bep),
+                              reinterpret_cast<std::complex<double> *>(Kep));
+}
+
+int it1j0y0_wrap(double x, double *j0int, double *y0int) {
+  return special::it1j0y0_wrap(x, j0int, y0int);
+}
+
+int it2j0y0_wrap(double x, double *j0int, double *y0int) {
+  return special::it2j0y0_wrap(x, j0int, y0int);
+}
+
+int it1i0k0_wrap(double x, double *i0int, double *k0int) {
+  return special::it1i0k0_wrap(x, i0int, k0int);
+}
+
+int it2i0k0_wrap(double x, double *i0int, double *k0int) {
+  return special::it2i0k0_wrap(x, i0int, k0int);
+}
+
+int cfresnl_wrap(npy_cdouble z, npy_cdouble *zfs, npy_cdouble *zfc) {
+  return special::cfresnl_wrap({npy_creal(z), npy_cimag(z)},
+                               reinterpret_cast<std::complex<double> *>(zfs),
+                               reinterpret_cast<std::complex<double> *>(zfc));
```

Comment:
Indeed! This is exactly what concerned me originally, because it wasn't going to be possible to copy the entire array of `npy_complex` for variables like `zfs` and `zfc`. I have no idea why this is okay, but taking the address of a single variable and doing `reinterpret_cast` is not.

Reply:
I'm going to look into it, and maybe ask on Stackoverflow if I can't figure it out on my own. ChatGPT wasn't able to give a satisfactory answer either.